packages/core/src/model/nodes/CodeNode.ts

```diff
@@ -15,6 +15,9 @@ import { nodeDefinition } from '../NodeDefinition.js';
 
 export type CodeNode = ChartNode<'code', CodeNodeData>;
 
+const maskInput = (name: string) => name.trim().replace(/[^a-zA-Z0-9_]/g, '_');
```

Comment:
well technically `$` is valid in an identifier too, but it looks like valid identifiers are... [complicated](https://stackoverflow.com/a/9337047)

Reply:
I see this more as asking if Rivet should have validation rules for the name of inputs/outputs on nodes. As it is now, it's just a `stringList` type and you can put anything. That could be a headache later for someone, for example, making a plugin but doesn't realized "#$(#@" is a valid input name and it crashes their plugin.

My recommendation, is to introduce a new type as example "inputList" which includes validation rules. (i.e. a blank string is not a valid input name). In this way, maybe we can add an input list editor that would highlight that input as invalid, and again highlight it if the user enters an invalid name.

I say this, because I think there are other nodes with the same problem. Like the passthrough node might be an example. Anywhere, that the user can add/remove inputs/outputs manually is subject to these kinds of issues.

Reply:
That would be great,